zipp/__init__.py

```diff
@@ -378,9 +367,9 @@ def stem(self):
     def filename(self):
         return pathlib.Path(self.root.filename).joinpath(self.at)
 
-    def read_text(self, *args, **kwargs):
-        encoding, args, kwargs = _extract_text_encoding(*args, **kwargs)
-        with self.open('r', encoding, *args, **kwargs) as strm:
+    def read_text(self, encoding=None, errors=None, newline=None):
+        encoding = text_encoding(encoding)
+        with self.open('r', encoding, errors, newline) as strm:
```

Comment:
I wonder if this logic is still needed after removing the PyPy workaround. After all, if there are no stack level concerns, maybe the check in `self.open` is sufficient.